[pytorch] add script to run all codegen

ljk53 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #46978 [pytorch] continue to rewrite gen_python_functions.py with typed models
* #46977 [pytorch] get rid of cpp_type_str from pybind codegen
* #46976 [pytorch] simplify tensor options logic in pybinding codegen
* **#46243 [pytorch] add script to run all codegen**

Add util script to test whether any codegen output changes.

Differential Revision: [D24388873](https://our.internmc.facebook.com/intern/diff/D24388873)

ljk53 committed:
[pytorch] add script to run all codegen

Add util script to test whether any codegen output changes.

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] add script to run all codegen"


Add util script to test whether any codegen output changes.

[ghstack-poisoned]

ezyang commented on tools/autograd/gen_annotated_fn_args.py:
```diff
@@ -20,6 +20,7 @@
     get_py_variable_methods,
     op_name,
 )
+import argparse
```
Hm?

ljk53 replied:
Right - this codegen cannot run as standalone script without this import. And I noticed the other two codegens' main() was broken recently, as well.

@ezyang - shall we simply remove the main() from all the children scripts, as they are only supposed to be called from one of the master script? 
It might be useful for debugging purpose but we should add CI as you suggested. I moved the test script to the CI folder - what's the recommended entry point for lightweight smoke tests like this?

ezyang replied:
IIRC, we have to run codegen before we can run stuff like clang-format/clang-tidy. Swapping the logic in those CI jobs to use this script might be a simple way to smoke test it.

I don't think it's necessarily bad to support running scripts individually; can be helpful if you're only working on codegen in one file. I kind of agree there are too many entry points today though...

ezyang reviewed (approved):
A little torn about this. This script is obviously useful (and I would have definitely implemented this myself), but it also comes at the cost of being yet another entry point to the codegens (one that isn't being tested by CI, to wit).

On the plus side, I can easily see a future universe where we just have one consolidated entrypoint (e.g., `tools.codegen.gen`) so I guess this is no problem.

ljk53 committed:
Update on "[pytorch] add script to run all codegen"


Add util script to test whether any codegen output changes.

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] add script to run all codegen"


Add util script to test whether any codegen output changes.

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] add script to run all codegen"


Add util script to test whether any codegen output changes.

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] add script to run all codegen"


Add util script to test whether any codegen output changes.

Differential Revision: [D24388873](https://our.internmc.facebook.com/intern/diff/D24388873)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] add script to run all codegen"


Add util script to test whether any codegen output changes.

Differential Revision: [D24388873](https://our.internmc.facebook.com/intern/diff/D24388873)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] add script to run all codegen"


Add util script to test whether any codegen output changes.

Differential Revision: [D24388873](https://our.internmc.facebook.com/intern/diff/D24388873)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] add script to run all codegen"


Add util script to test whether any codegen output changes.

Differential Revision: [D24388873](https://our.internmc.facebook.com/intern/diff/D24388873)

[ghstack-poisoned]
